go-controller/pkg/metrics/ovn_db.go

```diff
@@ -349,7 +349,7 @@ func getOvnDbVersionInfo() {
 	}
 }
 
-func RegisterOvnDBMetrics(clientset kubernetes.Interface, k8sNodeName string) {
+func RegisterOvnDBMetrics(registry *prometheus.Registry, clientset kubernetes.Interface, k8sNodeName string) {
```

Comment:
I _think_ if you change the first argument to prometheus.Registerer you can get rid of the cast here https://github.com/ovn-org/ovn-kubernetes/pull/2027/files#diff-d3e705641cf316568685f42242d21ce326a100c063c657fd073fb5e9cafb0920R288 and use directly `prometheus.DefaultRegisterer`

Reply:
@fedepaol If I make it a prometheus.Registerer, seem to be hitting an issue with mustRegister()
`registry.MustRegister undefined (type *prometheus.Registerer is pointer to interface, not interface)`

Also, in some cases ovnRegistry (of type prometheus.Registry()) is passed to RegisterOvnDBMetrics(). 

Let me know if it make sense to retain the type of first argrument as prometheus.Registry().